fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/indexes/StandardIndexMaintainer.java

```diff
@@ -365,7 +365,7 @@ protected <M extends Message> CompletableFuture<Void> updateOneKeyAsync(@Nonnull
                 state.store.countKeyValue(FDBStoreTimer.Counts.DELETE_INDEX_KEY, FDBStoreTimer.Counts.DELETE_INDEX_KEY_BYTES, FDBStoreTimer.Counts.DELETE_INDEX_VALUE_BYTES,
                         keyBytes, valueBytes);
             }
-            if (state.store.isIndexWriteOnly(state.index) && state.index.isUnique()) {
+            if (isWriteOnlyOrUniquePending() && state.index.isUnique()) {
```

Comment:
I know we support unique indexes for rank indexes, am I correct in my understanding that rank indexes will inherit this change? Should other indexes inherit this change?

Reply:
The standard maintainer seems to be the only entity that adds/clears uniqueness violation items.

Reply:
In particular, this doesn't need to go into other index maintainers (at least not the ones defined in the Record Layer) because they all inherit from this maintainer. That includes the `RankIndexMaintainer`, for example.

This might actually be another argument for using the format version, though. In particular, if a user has a custom index maintainer implementation that does not inherit from `StandardIndexMaintainer`, they may have code they need to update regarding how uniqueness constraints and index state interact, and that might be more cognizant of the need to make that change if they also need to update the format version. (For example, they can guard their calls to `.setFormatVersion` to a maximum supported format version, and only update that number once they've updated their index maintainers.)